Modules/@babylonjs/react-native/shared/BabylonNative.cpp

```diff
@@ -105,10 +104,15 @@ namespace BabylonNative
                 });
             });
 
-            m_jsDispatcher([this]()
+            if (!m_isRenderingEnabled)
             {
-                m_resolveInitPromise();
-            });
+                m_jsDispatcher([this]()
+                    {
+                        m_resolveInitPromise();
+                        CreateInitPromise();
```

Comment:
A new promise is created after it's been resolved instead of when view is reset. Also, it's only done once thanks to `m_isRenderingEnabled` check.